flatfs.go

```diff
@@ -388,6 +388,16 @@ func (fs *Datastore) Put(key datastore.Key, value []byte) error {
 	return err
 }
 
+func (fs *Datastore) Sync(prefix datastore.Key) error {
+	fs.shutdownLock.RLock()
+	defer fs.shutdownLock.RUnlock()
+	if fs.shutdown {
+		return ErrClosed
+	}
+
+	return nil
```

Comment:
Is this the approach we want here? I see that FlatFS has support for both sync and async operation (async seems to be the default, but IPFS uses the sync version by default).

Do we want to support some sort of sync here in the event that the datastore is running in async mode? If so is it important enough to do something now or should I just file an issue?

Reply:
Let's file an issue and leave this as-is for now.

 A correct implementation would need to call fsync on all changed files. Doing so would require keeping track of file changes (or relying on timestamps).